compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -917,8 +917,7 @@ private BType resolveTypeDesc(BLangArrayType td, ResolverData data) {
                     Name pkgAlias = names.fromIdNode(sizeReference.pkgAlias);
                     Name typeName = names.fromIdNode(sizeReference.variableName);
 
-                    BSymbol sizeSymbol =
-                            symResolver.lookupMainSpaceSymbolInPackage(size.pos, data.env, pkgAlias, typeName);
+                    BSymbol sizeSymbol = getSymbolOfVarRef(size.pos, data.env, pkgAlias, typeName);
```

Comment:
I don't think this is the correct fix. When resolving the type of `const int[ARRAY_SIZE] y = [1];`(type of `y`), We need to define all the symbols related to top level nodes, as I mentioned here. https://github.com/ballerina-platform/ballerina-lang/pull/41987#issuecomment-1917609827